src/cpp/rtps/builtin/discovery/participant/PDPServer2.cpp

```diff
@@ -761,6 +761,15 @@ bool PDPServer2::process_change_acknowledgement(
     // DATA(p|w|r) case
     if (c->kind == fastrtps::rtps::ChangeKind_t::ALIVE)
     {
+
+        // if change is servers DATA(P) it could be already acked by all and skip this step
```

Comment:
```suggestion
        // If then change is the server's DATA(p), it could be already acked by all. In that case, we can skip checking ack status on every reader proxy
```